### PR TITLE
Johans/related md

### PR DESCRIPTION
**Pull request by jstorteboom**

### CODE REVIEW

**Review by bart-vandervliet — APPROVED** [review]

File: Services/Masterdata/Services.Masterdata.EF/Services.Masterdata.EF.csproj [hunk 1]
```diff
@@ -1,8 +1,22 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
+  <ItemGroup>
+    <None Remove="RawSql\StoredProcedures\sp_GetMasterdataRecursiveTree.sql" />
```
Comment (dageorgero) [inline Services/Masterdata/Services.Masterdata.EF/Services.Masterdata.EF.csproj:4]: add here also the new stored procedure: sp_GetMasterdataFilterByRelatedMasterdata.sql

File: Services/Masterdata/Services.Masterdata.EF/Services.Masterdata.EF.csproj [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Migrations\" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="RawSql\StoredProcedures\sp_GetMasterdataRecursiveTree.sql">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
```
Comment (dageorgero) [inline Services/Masterdata/Services.Masterdata.EF/Services.Masterdata.EF.csproj:20]: add here also the new stored procedure: sp_GetMasterdataFilterByRelatedMasterdata.sql

File: Services/Masterdata/Services.Masterdata.Web/Controllers/MasterdataController.cs [hunk 3]
```diff
@@ -111,6 +111,22 @@ public async Task<ResultPagedListModel<string>> GetTags(string masterdataType, [
         var result = await _masterdataService.GetTags(masterdataType, queryModel);
         return result;
     }
+
+    //[HttpGet("{masterdataType}/{masterdata}/related")]
```
Comment (Johanh-od) [inline Services/Masterdata/Services.Masterdata.Web/Controllers/MasterdataController.cs]: No commented code

File: Services/Masterdata/Services.Masterdata.Web/Controllers/MasterdataController.cs [hunk 4]
```diff
@@ -52,7 +52,7 @@ public async Task<ResultPagedListModel<MasterdataTypeListModel>> Get([FromQuery]
     /// <param name="queryModel">The settings for paging, sorting and filtering.</param>
     /// <returns>A list of masterdatas belonging to a specific masterdata type.</returns>
     [HttpGet("{masterdataType}")]
-    public async Task<ResultPagedListModel<MasterdataModel>> GetMasterdata(string masterdataType, [FromQuery] QueryModel queryModel)
+    public async Task<ResultPagedListModel<MasterdataModel>> GetMasterdata(string masterdataType, [FromQuery] MasterdataQueryModel queryModel)
```
Comment (Johanh-od) [inline Services/Masterdata/Services.Masterdata.Web/Controllers/MasterdataController.cs]: Its better to let your controller return (I)ActionResult<T> and rewrite line 112 to return this.Ok(result)

Comment (Johanh-od) [reply Services/Masterdata/Services.Masterdata.Web/Controllers/MasterdataController.cs]: Same for the other controller endpoints

**Review by bart-vandervliet — APPROVED** [review]

**Pull request merged**